Mono.Addins/Mono.Addins/RuntimeAddin.cs

```diff
@@ -345,7 +355,7 @@ Type GetType_Expensive (string typeName, string assemblyName)
 
 			foreach (var kvp in loadedAssemblies) {
 				var assembly = kvp.Value;
-				if (string.IsNullOrEmpty (assemblyName) || assembly.FullName == assemblyName) {
+				if (string.IsNullOrEmpty (assemblyName) || assembly.GetName().Name == assemblyName) {
```

Comment:
Why are we choosing the short name here?

Reply:
Honestly, I don't remember. I'll take a look.

Reply:
It seems like I did this change because Util.TryParseTypeName() does return the short assembly name:  https://github.com/mono/mono-addins/blob/main/Mono.Addins/Mono.Addins.Database/Util.cs#L82, so without this change assembly names would never match. I guess I fixed this long time ago since I don't remember the issue. I wonder how this is affecting VS Mac.

Reply:
AssemblyQualifiedName should give us a full name for the assembly name.

```
typeof(int).AssemblyQualifiedName
System.Int32, System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e
```

I think the problem is that we're wrapping it in a new AssemblyName and returning the name, because we should be able to use the fullname here.

Reply:
Basically, the part that we're passing to new AssemblyName is `System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e`. I'm wondering why we drop that, because we should preserve that information. If we want to have multiple AssemblyLoadContexts available, we _need_ that information.